Backpack/Gradient/Classes/BPKGradient.h

```diff
@@ -1,26 +1,112 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
 #import <Foundation/Foundation.h>
 
+/**
+ * These constants represent the directions available for gradients in Backpack.
+ * The direction specifies the location that the last color in the gradient will fall on. 
+ */
 typedef NS_ENUM(NSUInteger, BPKGradientDirection) {
+
+    /**
+     * Gradient runs from bottom to top.
+     */
     BPKGradientDirectionUp,
+
+    /**
+    * Gradient runs from top to bottom.
+    */
     BPKGradientDirectionDown,
+
+    /**
+    * Gradient runs from right to left.
+    */
     BPKGradientDirectionLeft,
+
+    /**
+    * Gradient runs from left to right.
+    */
     BPKGradientDirectionRight,
+
+    /**
+    * Gradient runs from bottom-right to top-left.
+    */
     BPKGradientDirectionTopLeft,
+
+    /**
+    * Gradient runs from bottom-left to top-right.
+    */
     BPKGradientDirectionTopRight,
+
+    /**
+     * Gradient runs from top-right to bottom-left.
+     */
     BPKGradientDirectionBottomLeft,
+
+    /**
+    * Gradient runs from top-left to bottom-right.
+    */
     BPKGradientDirectionBottomRight,
 };
 
 NS_ASSUME_NONNULL_BEGIN
 @class UIColor;
+
+/**
+ * The `BPKGradient` class contains definitions for the Backpack gradients.
+ * It has factory methods for creating instances of supported gradients.
+ * Instances of BPKGradient capture the configuration of the gradient, but
+ * are not concerned with rendering.
+ * @warning Don't render Backpack gradients with `CAGradientLayer`
+ * as it does not accurately render gradients for non-square views.
+ */
 @interface BPKGradient: NSObject
+
+/**
+ * The colors that define the gradient.
+ */
 @property (copy, nonatomic, readonly) NSArray<UIColor *> *colors;
+
+/**
```

Comment:
This bit needs to mention that the values in the array are `CGPoint`s encoded as `NSValue`s using `valueWithPoint:` and the consumer can get the values back with the `CGPointValue` 

```objective-c
stops[0].GCPointValue
```